rust/transaction_parsing/src/lib.rs

```diff
@@ -56,15 +58,44 @@ where
     }
 
     match &data_hex[4..6] {
-        "00" | "02" => parse_transaction(data_hex, db_path),
+        "00" | "02" => parse_transaction(data_hex, db_path, false),
         "03" => process_message(data_hex, db_path),
+        "04" => parse_transaction_bulk(data_hex, db_path),
         "80" => load_metadata(data_hex, db_path),
         "81" => load_types(data_hex, db_path),
         "c1" => add_specs(data_hex, db_path),
         "de" => process_derivations(data_hex, db_path),
         _ => Err(Error::PayloadNotSupported(data_hex[4..6].to_string())),
     }
 }
+fn parse_transaction_bulk<P: AsRef<Path>>(payload: &str, dbname: P) -> Result<TransactionAction> {
+    let decoded_data = unhex(payload)?;
+
+    let bulk = TransactionBulk::decode(&mut &decoded_data[3..])?;
+
+    match bulk {
+        TransactionBulk::V1(b) => {
+            let mut actions = vec![];
+
+            let mut checksum = 0;
+            for t in &b.encoded_transactions {
+                let encoded = hex::encode(t);
+                let encoded = "53".to_string() + &encoded;
+                let action = parse_transaction(&encoded, &dbname, true)?;
+                if let TransactionAction::Sign {
```

Comment:
If bulk transactions can e any transactions, shouldn't we handle all transaction actions here?

Reply:
So at the moment we are suppose that a bulk has a no-nonsense content and contains only `Sign` variant and ignores the rest. But I want to reduce the surface of the changeset and keep the code as brief as possible by re-using the existing solutions, so no specific new variant is added to the types, just the existing variants are enhanced to contain a set of `Sign` transactions instead of one if that makes sense.

Reply:
Ok, then We need to add this check on platform code and probably to UI, that user can understand that multisign should work only for signing transactions. cc @tetekinandrey